src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationFrameworkValueProvider.cs

```diff
@@ -190,5 +240,114 @@ private static async Task<bool> IsWPFApplicationAsync(IProjectProperties default
+            };
+
+            if (propertyName == AuthenticationModeProperty)
+            {
+                value = value switch
+                {
+                    "0" => "Windows",
+                    "1" => "ApplicationDefined",
+                    "" => "",
+
+                    _ => throw new InvalidOperationException($"Invalid value '{value}' for '{propertyName}' property.")
+                };
+            }
+            else if (propertyName == HighDpiModeProperty)
+            {
+                value = value switch
+                {
+                    "0" => "DpiUnaware",
+                    "1" => "SystemAware",
+                    "2" => "PerMonitor",
+                    "3" => "PerMonitorV2",
+                    "4" => "DpiUnawareGdiScaled",
+                    "" => "",
+
+                    _ => throw new InvalidOperationException($"Invalid value '{value}' for '{propertyName}' property.")
+                };
+            }
+            else if (propertyName == ShutdownModeProperty)
+            {
+                value = value switch
+                {
+                    "0" => "AfterMainFormCloses",
+                    "1" => "AfterAllFormsClose",
+                    "" => "",
+
+                    _ => throw new InvalidOperationException($"Invalid value '{value}' for '{propertyName}' property.")
+                };
+            }
+
+            return value;
+        }
+
+        private async Task<string?> SetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties)
+        {
+            // ValueProvider needs to convert string enums to valid values to be saved.
+            if (propertyName == AuthenticationModeProperty)
+            {
+                unevaluatedPropertyValue = unevaluatedPropertyValue switch
+                {
+                    "Windows" => "0",
+                    "ApplicationDefined" => "1",
+                    _ => unevaluatedPropertyValue
+                };
+            }
+            else if (propertyName == HighDpiModeProperty)
+            {
+                unevaluatedPropertyValue = unevaluatedPropertyValue switch
+                {
+                    "DpiUnaware" => "0",
+                    "SystemAware" => "1",
+                    "PerMonitor" => "2",
+                    "PerMonitorV2" => "3",
+                    "DpiUnawareGdiScaled" => "4",
+                    _ => unevaluatedPropertyValue
+                };
+            }
+            else if (propertyName == ShutdownModeProperty)
+            {
+                unevaluatedPropertyValue = unevaluatedPropertyValue switch
+                {
+                    "AfterMainFormCloses" => "0",
+                    "AfterAllFormsClose" => "1",
+                    _ => unevaluatedPropertyValue
+                };
+            }
+
+            await(propertyName switch
```

Comment:
nit: this looks a bit like an invocation because of the `(`. Could be `await propertyName switch ...` instead.